gcloud-java-resourcemanager/src/test/java/com/google/gcloud/resourcemanager/ProjectTest.java

```diff
@@ -0,0 +1,214 @@
+    List<Member> viewerMemberList = new ArrayList<>();
+
+    ownerMemberList.add(Member.user("first-owner@email.com"));
+    ownerMemberList.add(Member.group("group-of-owners@email.com"));
+    OWNER_BINDING = Policy.Binding.builder().role(RoleType.OWNER).members(ownerMemberList).build();
+    editorMemberList.add(Member.serviceAccount("editor@someemail.com"));
+    EDITOR_BINDING =
+        Policy.Binding.builder().role(RoleType.EDITOR).members(editorMemberList).build();
+    viewerMemberList.add(Member.serviceAccount("app@someemail.com"));
+    viewerMemberList.add(Member.user("viewer@email.com"));
+    VIEWER_BINDING =
+        Policy.Binding.builder().role(RoleType.VIEWER).members(viewerMemberList).build();
+  }
+  private static final Policy POLICY =
+      Policy.builder()
+          .addBinding(OWNER_BINDING)
+          .addBinding(EDITOR_BINDING)
+          .addBinding(VIEWER_BINDING)
+          .version(1)
+          .etag("some-etag-value")
+          .build();
+  private static final List<Permission> PERMISSIONS_REQUESTED =
+      Arrays.asList(new Permission[] {Permission.REPLACE, Permission.GET});
+  private static final List<Boolean> PERMISSIONS_OWNED = Arrays.asList(new Boolean[] {false, true});
+
+  private ResourceManager resourceManager;
+  private Project project;
+
+  @Before
+  public void setUp() throws Exception {
+    resourceManager = createStrictMock(ResourceManager.class);
+    project = new Project(resourceManager, PROJECT_INFO, POLICY);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    verify(resourceManager);
+  }
+
+  @Test
+  public void testLoad() {
+    expect(resourceManager.get(PROJECT_INFO.id())).andReturn(PROJECT_INFO);
+    expect(resourceManager.getIamPolicy(PROJECT_INFO.id())).andReturn(POLICY);
+    replay(resourceManager);
+    Project loadedProject = Project.load(resourceManager, PROJECT_INFO.id());
+    assertEquals(PROJECT_INFO, loadedProject.info());
+    assertEquals(POLICY, loadedProject.policy());
+  }
+
+  @Test
+  public void testReload() {
+    ProjectInfo newInfo = PROJECT_INFO.toBuilder().addLabel("k3", "v3").build();
+    Policy newPolicy = POLICY.toBuilder().removeBinding(VIEWER_BINDING).build();
+    expect(resourceManager.get(PROJECT_INFO.id())).andReturn(newInfo);
+    expect(resourceManager.getIamPolicy(PROJECT_INFO.id())).andReturn(newPolicy);
+    replay(resourceManager);
+    Project newProject = project.reload();
+    assertSame(resourceManager, newProject.resourceManager());
+    assertEquals(newInfo, newProject.info());
+    assertEquals(newPolicy, newProject.policy());
+  }
+
+  @Test
+  public void testPolicy() {
+    assertEquals(project.policy(), POLICY);
+    replay(resourceManager);
+  }
+
+  @Test
+  public void testInfo() {
+    assertEquals(project.info(), PROJECT_INFO);
+    replay(resourceManager);
+  }
+
+  @Test
+  public void testResourceManager() {
+    assertEquals(project.resourceManager(), resourceManager);
```

Comment:
Nit, usually the first parameter is the expected value and the second one the actual.